server/src/main/java/org/candlepin/async/tasks/JobCleaner.java

```diff
@@ -63,36 +66,66 @@ public JobCleaner(Configuration config, JobManager jobManager) {
 
     @Override
     public void execute(JobExecutionContext context) throws JobExecutionException {
-        String cfgName = ConfigProperties.jobConfig(JOB_KEY, CFG_MAX_JOB_AGE);
-        int maxAgeInMinutes = this.config.getInt(cfgName, CFG_DEFAULT_MAX_JOB_AGE);
+        String termCfgName = ConfigProperties.jobConfig(JOB_KEY, CFG_MAX_TERMINAL_JOB_AGE);
+        Date termCutoff = this.parseMaxJobAgeConfig(termCfgName, CFG_DEFAULT_MAX_TERMINAL_JOB_AGE);
+
+        String nontermCfgName = ConfigProperties.jobConfig(JOB_KEY, CFG_MAX_NONTERMINAL_JOB_AGE);
+        Date nontermCutoff = this.parseMaxJobAgeConfig(nontermCfgName, CFG_DEFAULT_MAX_NONTERMINAL_JOB_AGE);
+
+        int removed = this.cleanupTerminalJobs(termCutoff);
+        int aborted = this.abortNonTerminalJobs(nontermCutoff);
+
+        context.setJobResult(String.format(
+            "Removed %1$d terminal jobs older than %2$tF %2$tT%2$tz\n" +
+            "Aborted %3$d non-running jobs older than %4$tF %4$tT%4$tz",
+            removed, termCutoff, aborted, nontermCutoff));
+    }
 
-        if (maxAgeInMinutes < 1) {
-            String errmsg = String.format("Invalid value for max age, must be a positive integer: %s",
-                maxAgeInMinutes);
+    private Date parseMaxJobAgeConfig(String cfgName, int defaultValue) throws JobExecutionException {
+        int value = this.config.getInt(cfgName, defaultValue);
+
+        if (value < 1) {
+            String errmsg = String.format("Invalid value for %s, must be a positive integer: %s",
+                cfgName, value);
 
             log.error(errmsg);
             throw new JobExecutionException(errmsg, true);
         }
 
-        // Set cutoff (end) date to now - max age in minutes
-        Date cutoff = Util.addMinutesToDt(maxAgeInMinutes * -1);
+        return Util.addMinutesToDt(value * -1);
+    }
 
+    private int cleanupTerminalJobs(Date cutoff) {
         // We're targeting every terminal job
         Set<JobState> jobStates = Arrays.stream(JobState.values())
-            .filter(state -> state.isTerminal())
+            .filter(JobState::isTerminal)
             .collect(Collectors.toSet());
 
         // Build the query builder with our config
         AsyncJobStatusQueryBuilder queryBuilder = new AsyncJobStatusQueryBuilder()
             .setJobStates(jobStates)
             .setEndDate(cutoff);
 
-        int deleted = this.jobManager.cleanupJobs(queryBuilder);
+        int removed = this.jobManager.cleanupJobs(queryBuilder);
+        log.info("Removed %d terminal jobs older than %2$tF %2$tT%2$tz", removed, cutoff);
```

Comment:
This logging doesn't work. When the JobCleaner runs the logs print the literal string `Removed %d terminal jobs older than %2$tF %2$tT%2$tz`. The intellij tooltip says "More arguments provided (2) than placeholders specified (0)"

Reply:
Oh, whoops. That's a copy/paste mistake on my part.

Reply:
Fixed